internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -239,111 +275,138 @@ func (connector *SingleUseConnector) Connect(
-	clientLogin, err := connector.readLogin(clientBuffer)
-	if err != nil {
-		wrappedError := errors.Wrap(err, "failed to handle client login")
-		connector.sendError(wrappedError)
-		return nil, wrappedError
-	}
-
-	// Send the client login so that secretless can send honest client params to
-	// the server
-	clientLoginChannel <- clientLogin
+		connectionResultChan <-connectResult{
+			conn: driverConn,
+			err:  err,
+		}
+	} ()
 
-	// Block continuation until driver has completed connection
-	<-connectPhaseFinished
+	loginResp, backendConn, err := connector.waitForLoginResponse(connInterceptor, connectionResultChan)
 
 	if err != nil {
-		wrappedError := errors.Wrap(err, "failed to connect to mssql server")
-		connector.sendError(wrappedError)
-		return nil, wrappedError
+		connector.writeErrorToClient(err)
+		return nil, err
 	}
 
-	// TODO: 	Send the login response to the client 	 (#1016)
-	// TODO: 	Verify appropriate errors are passed to the client (#1013)
-	// TODO: 	Replace this with an actual 'ok' message from the server
-	//			once login completes within the driver
-	// TODO:    Rename this to "AuthenticationOKMessage"
-	if _, err = clientConn.Write(connector.CreateAuthenticationOKMessage()); err != nil {
+	if err = connector.writeLoginResponse(connector.clientBuff, *loginResp); err != nil {
 		wrappedError := errors.Wrap(
 			err,
 			"failed to send a successful authentication response to client",
 		)
-		connector.sendError(wrappedError)
+		connector.writeErrorToClient(wrappedError)
 		return nil, wrappedError
 	}
 
-	connector.backendConn = backendConn
-	return connector.backendConn, nil
+	return backendConn, nil
 }
 
-// TODO: Add ability to receive an MSSQL error and send it to the client (#1013)
-func (connector *SingleUseConnector) sendError(err error) {
-	//NOTE: no need to log, Secretless already does this for every error sent back
-	//from the Connect method
-	//connector.logger.Error(err)
-	connector.sendErrorToClient()
+func protocolError(err error) mssql.Error {
+	if _protocolErr, ok := errors.Cause(err).(mssql.Error); ok {
+		return _protocolErr
+	}
+
+	return mssql.Error{
+		// SQL Error Number - currently using 18456 (login failed for user)
+		// TODO: Find generic error number
+		Number:     18456,
+		// State -
+		// TODO: better understand this.
+		State:      0x01,
+		// Severity Class - 16 indicates a general error that can be corrected by the user.
+		Class:      16,
+		Message:    errors.Wrap(err, "secretless").Error(),
+		ServerName: "secretless",
+		ProcName:   "",
+		LineNo:     0,
+	}
 }
 
-// TODO: Add ability to receive an MSSQL error and send it to the client (#1013)
-func (connector *SingleUseConnector) sendErrorToClient() {
-	mssqlError := connector.CreateGenericErrorMessage()
-	if _, e := connector.clientConn.Write(mssqlError); e != nil {
-		connector.logger.Errorf("failed to write error %s to MSSQL client", e)
+func (connector *SingleUseConnector) waitForLoginResponse(
```

Comment:
I think we want a more accurate name here, since we're not merely waiting for a login response, we're performing a number of stages in the protocol sequence.  

Would `DoPreloginAndLoginSteps` be accurate?  Generally "and" names are a code smell, but it's nbd and that's something we can address later if we want when we refactor.  Here my only concern is that the method name accurately described what the method does.

Reply:
went for `WaitForServerConnection`. i think when the refactor takes place then there'll be a better arrangement but for now that method is essentially waiting